keps/sig-windows/2258-node-service-log-viewer/README.md

```diff
@@ -0,0 +1,377 @@
+`C:\var\log\`.
+
+The feature now enables the cluster administrator to interrogate all services.
+This could be prevented by having a whitelist of allowed services. But this
+comes with severe disadvantages as there could be nodes (especially with
+Windows) that have other services to support networking and monitoring.
+These services are variable and will depend on how the nodes have been
+configured. Here are some examples:
+- [hybrid-overlay-node](https://github.com/ovn-org/ovn-kubernetes/tree/master/go-controller/hybrid-overlay)
+- [windows-exporter](https://github.com/prometheus-community/windows_exporter).
+
+
+The `/var/log/` endpoint is enabled using the `enableSystemLogHandler` kubelet
+configuration options. To gain access to this new feature this option needs to
+be enabled. In addition when introducing this feature it will be hidden behind a
+`NodeLogs` feature gate in the kubelet that needs to be explicitly enabled. So
+you need to enable both options to get access to this new feature and disabling
+`enableSystemLogHandler` will disable the new feature irrespective of the
+`NodeLogs` feature gate.
+
+A reference implementation of this feature without the feature gate is
+available [here](https://github.com/kubernetes/kubernetes/pull/96120).
+
+#### kubectl
+
+`kubectl` has an existing `logs` command that is used to view the logs for a
+container in a pod or a specified resource. The sub-command looks at resource
+types, so can be extended to work with node objects to view the logs of services
+on the nodes. Given that the `logs` command depends on RBAC policies for access
+to appropriate resource type and associated endpoints, it will allow us to
+restrict node logs access to only cluster administrators as long as the cluster
+is setup in that manner. Access to the `node/logs` sub-resource needs to be
+explicitly granted as a user with access to `nodes` will not automatically have
+access to `node/logs`.
+
+The `logs` sub-command for node objects will follow a heuristics approach when
+asked to query for logs from a Windows or Linux service. If asked to get the
+logs from a service `foobar`, it will first assume `foobar` logs to the Linux
+journal / Windows eventing mechanisms (Application, System, and ETW). If unable
+to get logs from these, it will attempt to get logs from `/var/log/foobar.log`,
+`/var/log/foobar/foobar.log`, `/var/log/foobar*INFO` or
+`/var/log/foobar/foobar*INFO` in that order.
+Here are some examples and explanation of the options that will be added.
+```
+Examples:
+  # Show kubelet logs from all masters
+  kubectl logs nodes --role master -s kubelet
+
+  # Show docker logs from Windows nodes
+  kubectl logs nodes -l kubernetes.io/os=windows -s docker
+
+Options:
+      --case-sensitive=true: Filters are case sensitive by default. Pass --case-sensitive=false to do a case insensitive filter.
+  -g, --grep='': Filter log entries by the provided regex pattern. Only applies to node journal logs.
```

Comment:
Why do some flags only work on some implementations?  Can we at least clarify somewhere that "some capabilities are optional, and may not be supported by some sources of log information", rather than specifically "journal"?

Reply:
I will add a note to call this out during the implementation. Would that suffice or should I open a follow up PR to amend the enhancement.